web-console/future-openshift/resource-creation/resource-creation.md

```diff
@@ -0,0 +1,72 @@
+# Resource Creation
+
+## Overview
+* If a resource has a creation form (and YAML), the form will load on default.
+  * After the resource is created, users can edit the resource in YAML only.
+  * In the future, inline edit will be investigated as a way to allow users to edit fields in a form-like manner.
+* Samples available in YAML editor will show YAML code to be copied and pasted as well as a "Use this sample" button that will overwrite the YAML with the sample.
+* Samples available in form will show a "Use this sample" button and will fill in form fields.
+* If users switch from form view to YAML, they will not be able to switch back.
+  * In future, users may be allowed to switch back and forth using a toggle button.
+    * Changes made in YAML will be reflected when switching to form view, and vice versa.
+    * An inline notification should appear on the form view notifying users that any additional fields they add in YAML will be saved.
+
+
+## Pages
+The follow documents which pages should have a YAML / form switcher and a samples panel. This list may be updated over time to reflect changes to our resource creation flows.
+
+**YAML / form switcher**
+* Create Key/Value Secret
+* Create Image Pull Secret
+* Create Source Secret
+* Create Webhook Secret
+* Create Route
+* Create Persistent Volume Claim
+* Create Storage Class
+
+**Samples**
+* Create Build Config
+* Create Resource Quota
+
+**Both switcher and samples**
+* Create Network Policy
+* Create Role
+* Create Rule
+
+
+## Switcher
+![1](img/switcher-form.png)
```

Comment:
Even though we went away from the toggle for now, I still liked the secondary button styling rather than a plain link as it felt more grounded and less floating. Thoughts on leaving "Edit YAML" as a secondary button style?

Reply:
Agreed! I made it a link to be consistent with what we have now, but I also prefer the secondary button styling. Will update!